numba/core/config.py

```diff
@@ -193,10 +209,11 @@ def _readenv(name, ctor, default):
                 return default() if callable(default) else default
             try:
                 return ctor(value)
-            except Exception as e:
+            except Exception:
                 warnings.warn(f"Environment variable '{name}' is defined but "
                               f"its associated value '{value}' could not be "
-                              f"parsed.\nThe parse failed with exception: {e}.",
+                              "parsed.\nThe parse failed with exception:\n"
```

Comment:
Should this be removed? The exception often explains what the issue is, for example: 
https://github.com/numba/numba/blob/e56e93e0d65170746befec5f92c903b1cc364b5f/numba/core/config.py#L39-L41

Reply:
Since this is an warning, the exception is never shown. Without the traceback, it's hard to tell what's wrong. I added this because I ended with some a weird bug at one point and just looking at the messages is not useful for debugging.

Reply:
OK, the error string is rendered into the message, but if it's not useful then removing it seems fine.